dev/devenv-readme.md

```diff
@@ -12,49 +12,175 @@ First off, if you're on one of the following platforms, we provide the [devenv-i
+      <vendor>oracle</vendor>
+      <!-- Change the id value to match the jdk version on your system --> 
+      <id>jdk-8u31-linux-x64</id>
+    </provides>
+    <configuration>
+      <!-- Change the jdkHome value to point to the jdk location on your system --> 
+      <jdkHome>/home/myusername/workspaces/tools/jdk-8u31-linux-x64</jdkHome>
+    </configuration>
+  </toolchain>
 </toolchains>
 ```
 
+### Maven GPG Signing
+
+Given that most of this project's builds go through Maven Central, all builds for the project have signing turned on by default. If you do not have a GPG key configured on your system, you will receive errors like the following:
+
+```
+[INFO] --- maven-gpg-plugin:1.5:sign (sign-artifacts) @ bluebutton-parent ---
+gpg: directory `/home/karl/.gnupg' created
+gpg: new configuration file `/home/karl/.gnupg/gpg.conf' created
+gpg: WARNING: options in `/home/karl/.gnupg/gpg.conf' are not yet active during this run
+gpg: keyring `/home/karl/.gnupg/secring.gpg' created
+gpg: keyring `/home/karl/.gnupg/pubring.gpg' created
+gpg: no default secret key: secret key not available
+gpg: signing failed: secret key not available
+[INFO] ------------------------------------------------------------------------
+[INFO] BUILD FAILURE
+[INFO] ------------------------------------------------------------------------
+[INFO] Total time: 1.505 s
+[INFO] Finished at: 2016-06-13T09:06:37-04:00
+[INFO] Final Memory: 18M/481M
+[INFO] ------------------------------------------------------------------------
+[ERROR] Failed to execute goal org.apache.maven.plugins:maven-gpg-plugin:1.5:sign (sign-artifacts) on project bluebutton-parent: Exit code: 2 -> [Help 1]
+```
+
+To fix these errors, the simplest thing to do is just install GPG and create keys for yourself, per the instructions here: [Central Repository: Working with PGP Signatures](http://central.sonatype.org/pages/working-with-pgp-signatures.html).
+
+GPG signing can also be disabled by adding `-Dgpg.skip=true` to your Maven builds, e.g.:
+
+```
+$ mvn clean verify -Dgpg.skip=true
+```
+
+But please note that the `deploy` goal/phase will still fail if builds are not signed by an authorized user, as that's a requirement imposed by the repository itself.
+
+### Proper HAPI-FHIR branch
+
+Currently the Blue Button repositories require a specific version of the HAPI-FHIR libraries in order to build successfully.  The following commands should be run in order to configure the required libraries properly for the build:
+
+```
+$ git clone git@github.com:HHSIDEAlab/hapi-fhir.git
+$ cd hapi-fhir
+$ git checkout -b fix-race-condition-in-if-none-exists-2.4 origin/fix-race-condition-in-if-none-exists-2.4
+$ mvn clean install
+```
+
+### Skipping Tests
+
+The default install goal for the maven build will run the integration tests.  If you do not want to run them as some do use AWS resources use the following command line when executing the build:
+
+```
+$ mvn clean install -DskipITs
```

Comment:
Does that work? Interesting. I always use `-DskipITs=true`.

Reply:
Sent you a separate email on this as I cannot get either construct to skip some of the tests using aws creds in the data-pipeline.